File: dlna-dmc/src/main/java/com/android/cast/dlna/dmc/DLNACastService.java
package com.android.cast.dlna.dmc;

import android.content.Intent;

import com.orhanobut.logger.Logger;

import org.fourthline.cling.UpnpServiceConfiguration;
import org.fourthline.cling.android.AndroidUpnpServiceConfiguration;
import org.fourthline.cling.android.AndroidUpnpServiceImpl;
import org.fourthline.cling.android.FixedAndroidLogHandler;
import org.fourthline.cling.model.types.ServiceType;

/**
 *
 */
public class DLNACastService extends AndroidUpnpServiceImpl {

    @Override
    public void onCreate() {
        Logger.i(String.format("[%s] onCreate", getClass().getName()));
        org.seamless.util.logging.LoggingUtil.resetRootHandler(new FixedAndroidLogHandler());
        super.onCreate();
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        Logger.i(String.format("[%s] onStartCommand: %s , %s", getClass().getName(), intent, flags));
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public void onDestroy() {
        Logger.w(String.format("[%s] onDestroy", getClass().getName()));
        super.onDestroy();
    }

    @Override
    protected UpnpServiceConfiguration createConfiguration() {
        return new DLNACastServiceConfiguration();
    }

    // ----------------------------------------------------------------
    // ---- configuration
    // ----------------------------------------------------------------
    private static final class DLNACastServiceConfiguration extends AndroidUpnpServiceConfiguration {

        @Override
        public int getRegistryMaintenanceIntervalMillis() {
            return 5000; //default is 3000!
        }

        @Override
        public ServiceType[] getExclusiveServiceTypes() {
            return new ServiceType[]{
                    DLNACastManager.SERVICE_RENDERING_CONTROL,
                    DLNACastManager.SERVICE_AV_TRANSPORT,
                    DLNACastManager.SERVICE_CONNECTION_MANAGER,
                    DLNACastManager.SERVICE_CONTENT_DIRECTORY};
        }
    }
}

File: dlna-dmc/src/main/java/com/android/cast/dlna/dmc/DLNACastService.kt
package com.android.cast.dlna.dmc

import android.content.Intent
import com.android.cast.dlna.core.Logger
import org.fourthline.cling.UpnpServiceConfiguration
import org.fourthline.cling.android.AndroidUpnpServiceConfiguration
import org.fourthline.cling.android.AndroidUpnpServiceImpl
import org.fourthline.cling.model.types.ServiceType

class DLNACastService : AndroidUpnpServiceImpl() {
    private val logger = Logger.create("CastService")
    override fun onCreate() {
        logger.i("DLNACastService onCreate")
//        LoggingUtil.resetRootHandler(FixedAndroidLogHandler())
        super.onCreate()
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        logger.i("DLNACastService onStartCommand: $flags, $startId, $intent")
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onDestroy() {
        logger.w("DLNACastService onDestroy")
        super.onDestroy()
    }

    override fun createConfiguration(): UpnpServiceConfiguration = object : AndroidUpnpServiceConfiguration() {
        override fun getExclusiveServiceTypes(): Array<ServiceType> = arrayOf(
            DLNACastManager.SERVICE_TYPE_AV_TRANSPORT,
            DLNACastManager.SERVICE_TYPE_RENDERING_CONTROL,
            DLNACastManager.SERVICE_TYPE_CONTENT_DIRECTORY,
            DLNACastManager.SERVICE_CONNECTION_MANAGER
        )
    }
}